<?xml version="1.0"?>
<launch>
  <!-- The node to read waypoints and generate the path -->
  <node pkg="mrc_examples" type="rabbit.py" name="rabbit" output="screen">
    <param name="update_rate" type="double" value="10"/>
    <param name="velocity" type="double" value="3"/>
    <rosparam file="$(find mrc_examples)/config/rabbit_waypoints_marina.yaml" />
  </node>

  <!-- Spawn a visual model to serve as marker -->
  <node name="spawn_rabbit" pkg="gazebo_ros" type="spawn_model"
        args="-x 0 -y 0 -z 0 -file $(find mrc_examples)/urdf/rabbit.urdf
              -urdf -model rabbit"/>

  <!-- Convert PointStamped messages to model states -->
  <node name="pointstamped2modelstate" pkg="mrc_examples" type="pointstamped2modelstate.py" output="screen">
    <param name="model_name" value="rabbit"/>
    <param name="reference_frame" value="world"/>
  </node>
  
</launch>
